### PR TITLE
in user-facing contexts, call it REPL not interpreter

### PR DESCRIPTION
**Pull request by SethTisue**
"Interpreter" is antiquated and not-actually-accurate terminology. Code entered in the Scala REPL is compiled, not interpreted. The use of "interpreter" to refer to any programming language's interactive shell is a relic of an era when interpreted languages had REPLs and compiled languages didn't.

thx @deanwampler for noticing this

### CODE REVIEW

**Comment by SethTisue** [comment]
If this is merged, I'll check Scala 3 and see if it needs a similar update anywhere.

**Review by ddl-deanwampler — COMMENTED** [review]
LGTM!

**Comment by som-snytt** [comment]
it's the replesentation compiler.

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala [hunk 1]
```diff
@@ -512,7 +512,7 @@ class ILoop(config: ShellConfig, inOverride: BufferedReader = null,
    */
   def resetCommand(line: String): Unit = {
     def run(destructive: Boolean): Unit = {
-      echo("Resetting interpreter state.")
+      echo("Resetting REPL state.")
```
Comment (som-snytt) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala:515]: `"OK, starting over."`

Comment (SethTisue) [reply src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala:515]: Eternal Sunshine of the Spotless REPL

**Comment by SethTisue** [comment]
fwiw, Scala 3 is already on board:

```
% scala3
Starting scala3 REPL...
scala>  
```

**Comment by som-snytt** [comment]
Shouldn't merge this without first noting that the REPL interprets user input -- `val x` can be a local variable or a template member, for instance -- and if it is wrapped in an object, the JVM literally interprets it, where is the ticket about slow execution, oh well, here is a good paulpism anyway: "every open script bug is a consequence of the way it is wrapped". Oh, the Sethism on ticket 3821 is "lots of fiddly stuff can affect the JIT."

Note "literally interprets it" is not "interprets it literally", which is an oxymoron.

I wouldn't mind wearing the t-shirt with the Roy Lichtenstein print of Seth saying in the speech balloon: _lots_ of fiddly stuff can affect the _**JIT**_!

**Comment by SethTisue** [comment]
> the JVM literally interprets it 

Hmm, well, that depends on whether you think the Chinese Room Argument proves anything.

Also known for their fiddly stuff: Nero; and the Devil, if you believe Charlie Daniels.

**Comment by som-snytt** [comment]
Now I don't know if I read the original novel, "Call it REPL", or just saw the '50s film version, or the '90s version that intended to be true to its source material, or the Netflix series that was a filmed representation of the novel had it been written in the 2000s as a screenplay.

**Comment by som-snytt** [comment]
Just found this pun in a test: if a wrapper is a skin, then a stack of wrappers is "replstiltskin".

**Pull request merged**